directives/questionnaire.py

```diff
@@ -120,6 +122,8 @@ def run(self):
                 raise SphinxError(u'Number of fields to sample randomly should greater than zero.')
             data[u'fieldgroups'][0]['pick_randomly'] = pick_randomly
 
+        self.set_assistant_permissions(data)
```

Comment:
The modification to the data dict should be done before the `override` option (in conf.py) is handled because the `override` setting is supposed to be able to override other settings. Now the local `allow-assistant-viewing` value can not be overridden.